terms/syntheticfeatures_fcm.tex

```diff
@@ -0,0 +1,151 @@
+      \textit & \makecell[tl]{Content} & \makecell[tl]{The comment body definitely has a big relevance in the classification result, \\ however, can the whole sentence be easily understood by the algorithm as \\ it is?} \\
+      \bottomrule
+    \end{tabular}}
+    \label{tab:sf_origdataexplained}
+\end{table}
+\vspace{2mm}
+
+As one can see, there is very little input the machine learning algorithm can reliably take just from using the four features described above. One could easily realise this by asking oneself the following question (in plain English):
+
+\qquad \emph{How can I describe the components of a comment well enough to decide whether it is probably spam or ham?}
+
+One can therefore summarise this problem paradoxically as: \emph{Having enough data to solve the problem, but very little meta-data to actually understand it and solve it.}
+
+\section{Ways of solving the problem}
+A way of solving this problem is to apply a synthetic features approach, sometimes referred to as feature engineering. This is the generation of features derived from other existing features, in a way that can be more easily captured or understood by a machine learning algorithm \citep{LiFeatureEng}. It is a way of generating meta-data for the existing features in the original dataset.
+
+In essence, the idea is to look at every available feature and for each determine the following:
+
+\begin{itemize}
+    \item Does the feature contain more than one feature within it? If so, try exploding it into sub-features and test.
+    \item Does the feature contain too little information for any relevance, but could benefit from adding some context to it? If so, attempt at looking at other features that might be related, and produce new features as a result, and test.
+    \item For each of the above, the original feature(s) might not be relevant anymore and be entirely replaced by the newly generated synthetic features instead.
+\end{itemize}
+
+What or how an explosion of features or a composite of features is generated depends on the very specific nature of the components involved and there is no generic formula behind it that works without some additional specificness. For example, two pairs of geo-coordinates probably qualify in giving a distance feature, however the formula applied here is specific to the geo-coordinates domain.
+
+There is not a one-size-fits-all approach but rather it is more of an iterative approach with new synthetic features being outputted per iteration, following which one then assesses whether it is enough to generate a reliable machine learning model from the new features or not.
+
+Following below is a practical example of this technique, using the dataset described at the introduction of this chapter.
+
+\subsection{Analysing each feature} \index{synthetic-features!solving-problem-analysing-each-feature}
+
+\textbf{Video}
```

Comment:
what is this?  use \subsubsection dont invent your own titling.  Also remove all manual spacing (`\\`) see guideline 31 in README.

Reply:
`\subsubsection` is not supported by Tufte as its philosophy discourages such subsections. I attempted to use it anyway but it failed with the error:

```
Class tufte-book Error: 
(tufte-book)              \subsubsectionis undefined by this class. See Robert Bringhurst's _The Elements of Typographic Style_, section 4.2.2. \subsubsectionwas used
```

See:
* [Tufte issue 44](https://github.com/rstudio/tufte/issues/44)
* [Tufte issue 34](https://github.com/rstudio/tufte/issues/34)

On this note, I'm not sure who I can proceed on it and leaving it as is for now unless you can suggest otherwise.

Reply:
Fixed by converting to lists.